webui/react/src/pages/F_ExpList/F_ExperimentList.tsx

```diff
@@ -1,27 +1,69 @@
+import { ExclamationCircleOutlined } from '@ant-design/icons';
 import { Rectangle } from '@glideapps/glide-data-grid';
 import { Row } from 'antd';
 import SkeletonButton from 'antd/es/skeleton/Button';
 import { observable } from 'micro-observables';
-import React, { useCallback, useEffect, useRef, useState } from 'react';
+import React, { useCallback, useEffect, useMemo, useRef, useState } from 'react';
 import { useSearchParams } from 'react-router-dom';
 
 import Page from 'components/Page';
+import useModalExperimentMove from 'hooks/useModal/Experiment/useModalExperimentMove';
+import usePermissions from 'hooks/usePermissions';
 import useResize from 'hooks/useResize';
-import { searchExperiments } from 'services/api';
+import {
+  activateExperiments,
+  archiveExperiments,
+  cancelExperiments,
+  getExperiments,
+  killExperiments,
+  openOrCreateTensorBoard,
+  pauseExperiments,
+  unarchiveExperiments,
+} from 'services/api';
+import { V1BulkExperimentFilters, V1GetExperimentsRequestSortBy } from 'services/api-ts-sdk';
 import usePolling from 'shared/hooks/usePolling';
+import { RecordKey } from 'shared/types';
+import { ErrorLevel } from 'shared/utils/error';
 import usersStore from 'stores/users';
-import { ExperimentItem, Project } from 'types';
+import {
+  ExperimentAction as Action,
+  BulkActionError,
+  CommandResponse,
+  ExperimentItem,
+  Project,
+  ProjectExperiment,
+} from 'types';
+import { modal } from 'utils/dialogApi';
 import handleError from 'utils/error';
+import {
+  canActionExperiment,
+  getActionsForExperimentsUnion,
+  getProjectExperimentForExperimentItem,
+} from 'utils/experiment';
 import { Loadable, Loaded, NotLoaded } from 'utils/loadable';
+import { openCommandResponse } from 'utils/wait';
 
 import { defaultExperimentColumns } from './glide-table/columns';
 import GlideTable from './glide-table/GlideTable';
+import TableActionBar from './glide-table/TableActionBar';
 import { useGlasbey } from './glide-table/useGlasbey';
 
 interface Props {
   project: Project;
 }
 
+const batchActions = [
+  Action.OpenTensorBoard,
+  Action.Activate,
+  Action.Move,
+  Action.Pause,
+  Action.Archive,
+  Action.Unarchive,
+  Action.Cancel,
+  Action.Kill,
+  Action.Delete,
+];
+
```

Comment:
consider expressing this subset of the `Action` type as its own type to use in `sendBatchAction` and `submitBatchAction`. That will indicate the `default` case of the switch statement in `sendBatchAction` as unreachable, meaning we shouldn't need to return an empty promise there.
```suggestion
] as const;

type BatchAction = typeof batchActions[number];
```